helixbass/tsc-rust
# Type-check object literal declaration

In this PR:
- type-check basic object literal variable declaration against annotated interface type

To test:
If you run `cargo run tmp.tsx` against a source file containing eg:
```
interface Foo {
  bar: number
}

const a: Foo = {
  bar: true,
}
```
you should see a diagnostic with `message_text: "Type 'boolean' is not assignable to type 'number'."` debug-logged
If you run `cargo run tmp.tsx` against a source file containing eg:
```
interface Foo {
  bar: number
}

const a: Foo = {
  bar: 1,
}
```
you shouldn't see any diagnostics debug-logged
